src/query/execute_schema.c

```diff
@@ -4573,12 +4577,24 @@ do_rename_partition (MOP old_class, const char *newname)
 
   newlen = strlen (newname);
 
-  error = au_fetch_class (old_class, &smclass, AU_FETCH_READ, AU_SELECT);
+  error = au_fetch_class (old_class, &smclass, AU_FETCH_UPDATE, AU_ALTER);
   if (error != NO_ERROR)
     {
       goto end_rename;
     }
 
+  strncpy (expr_copy, smclass->partition->expr, DB_MAX_PARTITION_EXPR_LENGTH);
+  keycol = expr_copy + 7;	/* strlen ("SELECT") + empty_string = 7 */
+  keycol_end = strstr (keycol, " ");
```

Comment:
Column names may contain spaces.
I don't think a simple whitespace check will solve it.

create table t1 ([cond name] int primary key, c2 varchar)
SELECT [cond  name] FROM ~

Reply:
Thank you. I didn't think of a blank.
I'll think of another way.